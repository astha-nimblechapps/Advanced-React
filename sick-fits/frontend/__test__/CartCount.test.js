import CartCountComponent from "../components/CartCount";
import { shallow, mount } from "enzyme";
import toJSON from 'enzyme-to-json';

describe('<CartCount/>',() => {
    it('renders',() =>{
        shallow(<CartCountComponent count={10}/>)
    });

    it('match snapshot',() => {
        const wrapper = shallow(<CartCountComponent count={10}/>)
        expect(toJSON(wrapper)).toMatchSnapshot()
    });

    /**
     * Shallow() will go one level deep testing
     * Mount() will test browser environment
     */

    it('updates props', () => {
        const wrapper = shallow(<CartCountComponent count={50}/>)
        expect(toJSON(wrapper)).toMatchSnapshot()
        wrapper.setProps({ count: 10 });
        expect(toJSON(wrapper)).toMatchSnapshot()
    })
});