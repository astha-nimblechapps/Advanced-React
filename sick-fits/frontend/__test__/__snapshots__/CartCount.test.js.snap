// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartCount/> match snapshot 1`] = `
<CartCount__AnimatedStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="10"
      timeout={
        Object {
          "enter": 400,
          "exit": 400,
        }
      }
      unmountOnExit={true}
    >
      <CartCount__Dot>
        10
      </CartCount__Dot>
    </CSSTransition>
  </TransitionGroup>
</CartCount__AnimatedStyles>
`;

exports[`<CartCount/> updates props 1`] = `
<CartCount__AnimatedStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="50"
      timeout={
        Object {
          "enter": 400,
          "exit": 400,
        }
      }
      unmountOnExit={true}
    >
      <CartCount__Dot>
        50
      </CartCount__Dot>
    </CSSTransition>
  </TransitionGroup>
</CartCount__AnimatedStyles>
`;

exports[`<CartCount/> updates props 2`] = `
<CartCount__AnimatedStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="10"
      timeout={
        Object {
          "enter": 400,
          "exit": 400,
        }
      }
      unmountOnExit={true}
    >
      <CartCount__Dot>
        10
      </CartCount__Dot>
    </CSSTransition>
  </TransitionGroup>
</CartCount__AnimatedStyles>
`;
